tests/cases/compiler/moduleResolutionWithSymlinks_preserveSymlinks.ts

```diff
@@ -0,0 +1,24 @@
+// @noImplicitReferences: true
+
+// @traceResolution: true
+// @preserveSymlinks: true
+// @moduleResolution: node
+
+// @filename: /linked/index.d.ts
+// @symlink: /app/node_modules/linked/index.d.ts,/app/node_modules/linked2/index.d.ts
+export { real } from "real";
```

Comment:
Added a test to `moduleResolution.ts`. I think the compiler test is valuable though, since it covers some assumptions we have about the effects of resolving / not resolving symlinks.